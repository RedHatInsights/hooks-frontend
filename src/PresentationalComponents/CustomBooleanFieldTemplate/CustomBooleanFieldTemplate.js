import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox, FormGroup } from '@patternfly/react-core';

export class CustomBooleanFieldTemplate extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: props.formData };
    }

    state = {
        value: false
    };

    handleTextInputChange = value => {
        this.props.onChange(value);
        this.setState({ value });
    };

    render() {
        const { value } = this.state;

        return <FormGroup fieldId={ `custom-checkbox-${ this.props.name }` } isRequired={ this.props.required }>
            <Checkbox
                label={ this.props.schema.title }
                aria-label={ this.props.schema.title }
                defaultChecked={ value }
                onChange={ this.handleTextInputChange }
                id={ `custom-checkbox-${ this.props.name }` }
                key={ `custom-checkbox-${ this.props.name }-key` }
            />
        </FormGroup>;
    }
}

CustomBooleanFieldTemplate.propTypes = {
    formData: PropTypes.bool,
    required: PropTypes.bool,
    schema: PropTypes.object,
    name: PropTypes.string,
    onChange: PropTypes.func
};

export default CustomBooleanFieldTemplate;
