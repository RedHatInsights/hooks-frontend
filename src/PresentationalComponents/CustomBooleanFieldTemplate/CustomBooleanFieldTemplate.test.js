import toJson from 'enzyme-to-json';

import { CustomBooleanFieldTemplate } from './CustomBooleanFieldTemplate';

describe('CustomBooleanFieldTemplate', () => {
    const testProps = {
        formData: true,
        schema: {
            title: 'Test Input Field'
        },
        onChange: jest.fn(),
        name: 'test-input-field',
        required: true
    };

    it('expect to render', () => {
        const wrapper = shallow(
            <CustomBooleanFieldTemplate { ...testProps } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('expect to call onChange handler', () => {
        const wrapper = shallow(
            <CustomBooleanFieldTemplate { ...testProps } />
        );
        wrapper.find('Checkbox').simulate('change', true);
        expect(testProps.onChange).toHaveBeenCalledWith(true);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
