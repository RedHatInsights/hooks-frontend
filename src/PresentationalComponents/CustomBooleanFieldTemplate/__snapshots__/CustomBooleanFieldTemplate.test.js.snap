// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBooleanFieldTemplate expect to render 1`] = `
<FormGroup
  className=""
  fieldId="custom-checkbox-test-input-field"
  isInline={false}
  isRequired={true}
  isValid={true}
>
  <Checkbox
    aria-label="Test Input Field"
    className=""
    defaultChecked={true}
    id="custom-checkbox-test-input-field"
    isChecked={null}
    isDisabled={false}
    isValid={true}
    key="custom-checkbox-test-input-field-key"
    label="Test Input Field"
    onChange={[Function]}
  />
</FormGroup>
`;
